### PR TITLE
Add anyhow context in mithril-client/commands

### PR DESCRIPTION
**Pull request by dlachaumepalo**
## Content
This PR includes an update of mithril-client `commands` with `anyhow` context.

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Issue(s)
Relates to #798

### CODE REVIEW

File: mithril-client/src/commands/mithril_stake_distribution/download.rs [hunk 1]
```diff
@@ -25,21 +26,34 @@ pub struct MithrilStakeDistributionDownloadCommand {
...
             .download(
                 &self.artifact_hash,
                 &self.download_dir,
                 &config.get_string("genesis_verification_key")?,
             )
-            .await?;
+            .await
+            .with_context(|| {
+                format!(
+                    "Mithril Stake Distribution Service can not download and verify the stake distribution, artifact hash: '{}'",
```
Comment (jpraynaud) [inline mithril-client/src/commands/mithril_stake_distribution/download.rs:53]:
```suggestion
                    "Mithril Stake Distribution Service can not download and verify the artifact for hash: '{}'",
```

File: mithril-client/src/commands/mithril_stake_distribution/list.rs [hunk 2]
```diff
@@ -22,8 +23,13 @@ impl MithrilStakeDistributionListCommand {
...
         let service = dependencies_builder
             .get_mithril_stake_distribution_service()
-            .await?;
-        let lines = service.list().await?;
+            .await
+            .with_context(|| {
+                "Dependencies Builder can not get Mithril Stake Distribution Service"
+            })?;
+        let lines = service.list().await.with_context(|| {
+            "Mithril Stake Distribution Service can not get the list of stake distributions"
```
Comment (jpraynaud) [inline mithril-client/src/commands/mithril_stake_distribution/list.rs:31]:
```suggestion
            "Mithril Stake Distribution Service can not get the list of artifacts"
```

File: mithril-client/src/commands/snapshot/download.rs [hunk 3]
```diff
@@ -30,13 +31,26 @@ pub struct SnapshotDownloadCommand {
...
-        let snapshot_entity =
-            FromSnapshotMessageAdapter::adapt(snapshot_service.show(&self.digest).await?);
+        let snapshot_service = dependencies_builder
+            .get_snapshot_service()
+            .await
+            .with_context(|| "Dependencies Builder can not get Snapshot Service")?;
+        let snapshot_entity = FromSnapshotMessageAdapter::adapt(
+            snapshot_service.show(&self.digest).await.with_context(|| {
+                format!(
+                    "Snapshot Service can not get the details of the snapshot, digest: '{}'",
```
Comment (jpraynaud) [inline mithril-client/src/commands/snapshot/download.rs:49]:
```suggestion
                    "Snapshot Service can not get the snapshot for digest: '{}'",
```

File: mithril-client/src/commands/snapshot/download.rs [hunk 4]
```diff
@@ -46,7 +60,14 @@ impl SnapshotDownloadCommand {
             .download(
                 &snapshot_entity,
                 &self.download_dir,
-                &config.get_string("genesis_verification_key")?,
+                &config
+                    .get_string("genesis_verification_key")
+                    .with_context(|| {
+                        format!(
+                            "Snapshot Service can not download and verify the snapshot, digest: '{}'",
```
Comment (jpraynaud) [inline mithril-client/src/commands/snapshot/download.rs:67]:
```suggestion
                            "Snapshot Service can not download and verify the snapshot for digest: '{}'",
```

File: mithril-client/src/commands/snapshot/show.rs [hunk 5]
```diff
@@ -25,8 +26,16 @@ impl SnapshotShowCommand {
...
         let mut dependencies_builder = DependenciesBuilder::new(Arc::new(config));
-        let snapshot_service = dependencies_builder.get_snapshot_service().await?;
-        let snapshot_message = snapshot_service.show(&self.digest).await?;
+        let snapshot_service = dependencies_builder
+            .get_snapshot_service()
+            .await
+            .with_context(|| "Dependencies Builder can not get Snapshot Service")?;
+        let snapshot_message = snapshot_service.show(&self.digest).await.with_context(|| {
+            format!(
+                "Snapshot Service can not show the details of the snapshot, digest: '{}'",
```
Comment (jpraynaud) [inline mithril-client/src/commands/snapshot/show.rs:35]:
```suggestion
                "Snapshot Service can not show the snapshot for digest: '{}'",
```

**Review by jpraynaud — COMMENTED** [review]
Few suggestions below:

File: mithril-client/src/commands/snapshot/download.rs [hunk 6]
```diff
@@ -30,13 +31,26 @@ pub struct SnapshotDownloadCommand {
 impl SnapshotDownloadCommand {
     /// Command execution
     pub async fn execute(&self, config_builder: ConfigBuilder<DefaultState>) -> StdResult<()> {
-        let config_builder = config_builder.set_default("genesis_verification_key", "")?;
+        let config_builder = config_builder
+            .set_default("genesis_verification_key", "")
+            .with_context(|| {
```
Comment (ghubertpalo) [inline mithril-client/src/commands/snapshot/download.rs:36]: If this case does never occur then `unwrap`.

**Review by ghubertpalo — APPROVED** [review]

**Pull request merged**